Modules/IO/MeshBase/include/itkMeshIOBase.h

```diff
@@ -382,8 +382,8 @@ class ITKIOMeshBase_EXPORT MeshIOBase:public LightProcessObject
   itkGetConstMacro(NumberOfPointPixelComponents, unsigned int);
   itkSetMacro(NumberOfCellPixelComponents, unsigned int);
   itkGetConstMacro(NumberOfCellPixelComponents, unsigned int);
-  itkSetMacro(PointDimension, unsigned int);
-  itkGetConstMacro(PointDimension, unsigned int);
+  itkSetMacro(PointDimension, SizeValueType);
```

Comment:
I was going to suggest to use `SizeValueType` instead of `size_t`, because that what these sort of types are usually called in ITK. But now I see you also `SizeValueType` here.

What happens with warning if you turn `ITK_USE_64BITS_IDS` off? Is there a difference before/after?

Reply:
@dzenanz Thanks for your thoughtful review! Yes I did use ITK's `SizeValueType` for `PointDimension`, because the corresponding data member `m_PointDimension` is actually of type `SizeValueType`! Still I wonder: is there any fundamental difference between `itk::SizeValueType` and `std::size_t`?

> What happens with warning if you turn ITK_USE_64BITS_IDS off?

Haven't checked yet, but honestly my main goal is to fully support the 64-bit address space.  So I'm focusing now on building ITK with `ITK_USE_64BITS_IDS=ON`, and `sizeof(void*)` is 8 bytes. However, I certainly do not want to break any 32-bit user code!

Note that those MSVC "possible loss of data" warnings are disabled now on various places in ITK. So I had to manually "enable" them, locally.

This pull request only fixes a few of those warnings. There are many more! But I thought I'd better first see your reactions on this PR...!

Reply:
`SizeValueType` usually boils down to `size_t` on Windows, and `unsigned long` on Linux. See [itkIntTypes.h](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkIntTypes.h). But on 32-bit Windows `SizeValueType` boils down to `unsigned long`, which is essentially the same as `unsigned int`. Hence the recommendation to check 32-bit compilation.

Reply:
Thanks @dzenanz Is it safe to assume `sizeof(itk::SizeValueType) == sizeof(std::size_t)`?

Reply:
Not always. But for 64-bit compilation and `ITK_USE_64BITS_IDS=ON`, which is the most common, yes.

Reply:
Thanks again, @dzenanz  Personally I think it's complicated enough already to support multiple platforms, with different sizes for `long` and `size_t`. I would be in favor of requiring that `itk::SizeValueType` has the same size as `std::size_t`. Ideally I would think: `using SizeValueType = std::size_t`. Unless there is a very strong use case to have it differently!

I feel that the current complexity makes it very hard to fully support the 64-bit address space. I guess that's why warnings about "loss of data" have just been disabled.

By the way, my interest in this issue was triggered by a post at the [elastix](https://github.com/SuperElastix/elastix) forum, [Elastix crashes without any notice](https://groups.google.com/forum/#!category-topic/elastix-imageregistration/elastix/Q1e1b2UNIQI), by Philipp Heinrich. So users actually want to go beyond the 32 bits...!

Do you also think it would be a good idea to require that `itk::SizeValueType` has the same size as `std::size_t`?

Reply:
Yes, I think 32-bit platforms have lost relevance about 10 years ago, at least in medical imaging field. And we could go the route of having SizeValueType = size_t. But I think you will find out that it will require quite a bit of refactoring to get right and flush out all the warnings. If you decide to go that way, I am willing to help you. But we should probably have approval, or at least absence of objections from @thewtex, @hjmjohnson and @blowekamp.

Reply:
It is relatively recently that for windows 64 that ITK_USE64_BIT_IDS is ON by default. This was done for compatibility because a lot of code was written with the unsigned long type. 

This issue should be discussed further on discourse.

Reply:
I am very much in support of having "having SizeValueType = size_t".

Reply:
Niels, can you start a discourse discussion about the proposed reorganization of various `typedef`s? Matt should be back from SciPy tomorrow, so he will have a chance to pitch in into the discussion.

Reply:
@blowekamp 
> It is relatively recently that for windows 64 that ITK_USE64_BIT_IDS is ON by default.

Thanks, Bradley! Just for the record, this appears to be the commit that did it:

Revision: 8bb8c743267ab0a0ac4e7b14f1ec2baf4a0fdc80
Author: Bradley Lowekamp
Date: March 3, 2016
ENH: Use ITK_USE_64BITS_IDS for windows 64 by default

> Support 4GB+ images and data structures by default on Windows 64.

> Note: ITK_USE_64BITS_IDS still defaults to being off on non-windows
systems which continues the use of type "unsigned long" for sizes as
uint64_t and "unsigned long" may not be the same type one these
systems.

Reply:
@dzenanz wrote:
> Niels, can you start a discourse discussion about the proposed reorganization of various `typedef`s?

 Please have a look: [Could ITK drop ITK_USE_64BITS_IDS and just do SizeValueType = std::size_t?](https://discourse.itk.org/t/could-itk-drop-itk-use-64bits-ids-and-just-do-sizevaluetype-std-size-t/2053)